### DISCUSSION TITLE
Add a Java SDK to query data directly from the database

### DISCUSSION THREAD

**Discussion by houxinlin** [apache/skywalking#10742] [Ideas]
Add a JavaSDK, which can directly search data from the database when skywalking is not started or abnormal, such as service and trace

**Comment by lujiajing1126** [comment]
Thanks for opening the discussion. I have a question: Why don't use the native UI of a specific database to explore the data, e.g. Kibana for ES?

**Reply by houxinlin** [reply]
> Thanks for opening the discussion. I have a question: Why don't use the native UI of a specific database to explore the data, e.g. Kibana for ES?

Yes，Because some projects need to obtain skywalking data for analysis, an interface is needed to provide data. The existing HTTP interface is not very friendly. It will be much more convenient to provide an interface directly connected to the database.

**Comment by wu-sheng** [comment]
I would say this SDK is helpful, meanwhile, we don't have a repository in the upstream suitable for this SDk hosting.
Are you willing to maintain and release this on your own?
We could add yours as ecosystem project on the website once you have complete docs(English) and released.
If more people are using it, we could initial the donation process and host it officially at the ASF.

**Reply by houxinlin** [reply]
yes i can do

**Reply by wu-sheng** [reply]
It is better you could push your release on maven central.(Just suggestion)
When you have repo and docs ready, feel free to open PR to skywalking-website repo for adding your repository.

**Reply by houxinlin** [reply]
ok, understand

**Reply by houxinlin** [reply]
> It is better you could push your release on maven central.(Just suggestion) When you have repo and docs ready, feel free to open PR to skywalking-website repo for adding your repository.

There are two implementation methods, one is to integrate storage-plugin, such as storage-jdbc-hikaricp-plugin, but a day-based data table is added after version 9.5, and the plugin seems to be unable to be used alone, and the other is to use it manually jdbc to query, do you have a better way?

**Reply by wu-sheng** [reply]
If you ask for our suggestion, we always only recommend for using GraphQL or PromQL.
These are the only official support query protocols. And the compatibility could be kept as much as possible.
Of course, this is your own repository, you could make the call.

**Reply by wu-sheng** [reply]
If you want to query Elasticsearch or JDBC, that is highly relying on whether upstream has breaking changes.

**Reply by houxinlin** [reply]
> If you want to query Elasticsearch or JDBC, that is highly relying on whether upstream has breaking changes.

Yes, I know, but it makes some projects easier to write if  use JDBC and provide an interface.

**Reply by houxinlin** [reply]
> If you want to query Elasticsearch or JDBC, that is highly relying on whether upstream has breaking changes.

Thank you for your answer

**Reply by wu-sheng** [reply]
> > If you want to query Elasticsearch or JDBC, that is highly relying on whether upstream has breaking changes.
> 
> 
> 
> Yes, I know, but it makes some projects easier to write if  use JDBC and provide an interface.

Your project, your call. 
You don't need upstream permission. 
You need to persuade end users about this is a good way. An open source project has meaning only when it has users.

**Reply by houxinlin** [reply]
ok, understand
